text/entities/0256-entities-data-model.md

```diff
@@ -0,0 +1,790 @@
+</td>
+   </tr>
+</table>
+
+Note that transmitting EntityDelete is not guaranteed
+when the entity is gone. Recipients of entity signals need to be prepared to
+handle this situation by expiring entities that are no longer seeing
+EntityState events reported (i.e. treat the presence of EntityState
+events as a liveliness indicator).
+
+The expiration mechanism is based on the previously reported `Interval` field of
+EntityState event. The recipient can use this value to compute when to expect the next
+EntityState event and if the event does not arrive in a timely manner (plus some slack)
+it can consider the entity to be gone even if the EntityDelete event was not observed.
+
+## Entity Identification
+
+_This section is a supplementary guideline and is not part of logical data model._
+
+The data model defines the structure of the entity Id field. This section explains
+how the Id field is computed.
+
+### LID, GID and IDCONTEXT
+
+All entities have a local ID (LID) and a global ID (GID).
+
+The LID is unique in a particular identification context, but is not necessarily globally
+unique. For example a process entity's LID is its PID number and process start time.
+The (PID,StartTime) pair is unique only in the context of a host where the process runs
+(and the host in this case is the identification context).
+
+The GID of an entity is globally unique, in the sense that for the entire set of entities
+in a particular telemetry store no 2 entities exist that have the same GID value.
+
+The GID of an entity E is defined as:
+
+`GID(E) = UNION( LID(E), GID(IDCONTEXT(E)) )`
+
+Where `IDCONTEXT(E)` is the identification context in which the LID of entity E is unique.
+The value of `IDCONTEXT(E)` is an entity itself, and thus we can compute the GID value of it too.
+
+In other words, the GID of an entity is a union of its LID and the GID of its
+identification context. Note: GID(E) is a map of key/value attributes.
+
+The enrichment process often is responsible for determining the value of `IDCONTEXT(E)`
+and for computing the GID according to the formula defined above, although the GID may
+also be produced at once by the telemetry source (e.g. by OTel SDK) without requiring
+any additional enrichment.
+
+### Semantic Conventions
+
+OpenTelemetry semantic conventions will be enhanced to include entity definitions for
+well-known entities such as Service, Process, Host, etc.
+
+For well-known entity types LID(E) is defined in OTel semantic conventions per
+entity type. The value of LID is a map of key/value attributes. For example,
+for entity of type "process" the semantic conventions define LID as 2 attributes:
+
+```json5
+{
+  "process.pid": $pid,
+  "process.start_time": $starttime
+}
+```
+
+For custom entity types (not defined in OTel semantic conventions) the end-user is
+responsible for defining their custom semantic conventions in a similar way.
+
+The entity information producer is responsible for determining the identification
```

Comment:
How should the service know it's running in a k8S cluster for example?

Reply:
The service likely won't know. It will be the job of the Collector to figure it out. See this section that explains how the Collector can enrich data emitted by Otel SDK: https://docs.google.com/document/d/1VUdBRInLEhO_0ABAoiLEssB1CQO_IcD5zDnaMEha42w/edit#heading=h.ij8yjheo645z

Reply:
thanks